File: backend/app/schemas/articles.py
"""Article-related schemas."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field, field_validator


class ArticleSummaryItem(BaseModel):
    """Article row used in overview and list pages."""

    id: int
    title: str
    slug: str
    quality_score: int
    publish_status: int
    dim_subject: str | None = None
    dim_action: str | None = None
    dim_attribute: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class ArticleDetailItem(ArticleSummaryItem):
    """Detailed article payload."""

    meta_json: dict[str, Any] | list[Any] | None = None
    content_markdown: str | None = None
    target_keywords: list[str] = Field(default_factory=list)
    outgoing_links_count: int = 0
    incoming_links_count: int = 0
    related_run_id: int | None = None
    related_run_status: str | None = None


class ArticleListPayload(BaseModel):
    """Paginated article list."""

    items: list[ArticleSummaryItem] = Field(default_factory=list)
    total: int
    limit: int
    offset: int
    warning: str | None = None


class ArticleSummaryPayload(BaseModel):
    """Aggregated article counts."""

    total_articles: int
    draft_articles: int
    approved_articles: int
    published_articles: int
    average_quality_score: float | None = None
    warning: str | None = None


class ArticleDetailPayload(BaseModel):
    """Single article response."""

    article: ArticleDetailItem | None = None
    warning: str | None = None


class ArticlePublishRequest(BaseModel):
    """Request body for manual article publishing."""

    platforms: list[str] = Field(default_factory=list)
    go_live: bool = False

    @field_validator("platforms")
    @classmethod
    def validate_platforms(cls, value: list[str]) -> list[str]:
        allowed = {"zhihu", "wechat"}
        cleaned = []
        for item in value:
            normalized = str(item).strip().lower()
            if normalized in allowed and normalized not in cleaned:
                cleaned.append(normalized)
        return cleaned

File: frontend_v2/src/types/articles.ts
import type { PaginatedPayload } from './api'

export interface ArticleSummaryItem {
  id: number
  title: string
  slug: string
  quality_score: number
  publish_status: number
  dim_subject: string | null
  dim_action: string | null
  dim_attribute: string | null
  created_at: string | null
  updated_at: string | null
}

export interface ArticleDetailItem extends ArticleSummaryItem {
  meta_json: Record<string, unknown> | unknown[] | null
  content_markdown: string | null
  target_keywords: string[]
  outgoing_links_count: number
  incoming_links_count: number
  related_run_id: number | null
  related_run_status: string | null
}

export type ArticleListPayload = PaginatedPayload<ArticleSummaryItem>

export interface ArticleSummaryPayload {
  total_articles: number
  draft_articles: number
  approved_articles: number
  published_articles: number
  average_quality_score: number | null
  warning: string | null
}

export interface ArticleDetailPayload {
  article: ArticleDetailItem | null
  warning: string | null
}

export interface ArticleListFilters {
  status?: string
  min_score?: number
  query?: string
  limit?: number
  offset?: number
}

export interface ArticlePublishRequest {
  platforms: Array<'zhihu' | 'wechat'>
  go_live: boolean
}

export type ArticleMutationResult = Record<string, unknown>
